src/ML-METATENSOR/pair_metatensor.h

```diff
@@ -0,0 +1,73 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+#ifdef PAIR_CLASS
+// clang-format off
+PairStyle(metatensor, PairMetatensor);
+// clang-format on
+#else
+
+#ifndef LMP_PAIR_METATENSOR_H
+#define LMP_PAIR_METATENSOR_H
+
+#include <memory>
+
+#include "pair.h"
+
+#include <metatensor/torch/atomistic.hpp>
+
+namespace LAMMPS_NS {
+class MetatensorSystemAdaptor;
+
+class PairMetatensor : public Pair {
+public:
+    PairMetatensor(class LAMMPS *);
+    ~PairMetatensor();
+
+    void compute(int, int) override;
+    void settings(int, char **) override;
+    void coeff(int, char **) override;
+    void init_style() override;
+    double init_one(int, int) override;
+    void init_list(int id, NeighList *ptr) override;
+
+    void allocate();
+
+private:
+    void load_torch_model(const char* path, const char* extensions_directory);
+
```

Comment:
The details of the implementation, if they need classes and data types outside of the ones used by LAMMPS itself, should be hidden and the header "PIMPL"-ified, i.e. the header should include just a pointer to a struct and then the struct itself is defined in the corresponding .cpp file. Please see pair_pace.h and pair_pace.cpp in the ML-PACE package for an example where it uses `struct ACEImpl *aceimpl;` in the header.